### ISSUE TITLE
If tsh cli requests with labels fails midway, it stops the request completely not continuing with remaining nodes

### ISSUE THREAD

**Issue by TeleLos** [gravitational/teleport#31579]
Expected behavior:
User executes **tsh ssh root@env=aws hostname** the command should run hostname on all nodes with the 'env: aws' label. 

Current behavior:
The request detects that I have Multiple noes matching the label selector and attempts to run the command on all. However, if one of the nodes is off-line the request stops and doesn't run on any other nodes matching the label selector. The problem is that the off-line node encountered midway the request doesn't run an the remaining nodes. 

Bug details:
- Teleport Proxy version: 13.3.7, tsh version 13.3.6. 
- Recreation steps

1.   Apply a common label to multiple nodes. <key=value> 
2.   Execute command: tsh ssh root@<key=value> hostname
3.   Determine the order in which the nodes are checked. 
4.   Stop the teleport.service on the node that matches the criteria but falls in the middle
5.   Run step 2 again. 

- Debug logs/error reported

```
WARNING: Multiple nodes matched label selector, running command on all.
Running command on 292d9306:
Thu 07 Sep 2023 03:09:57 PM UTC
failed connecting to host <host_id>:0: failed to dial target host
	Teleport proxy failed to connect to "node" agent "@local-node" over reverse tunnel:

  no tunnel connection found: no node reverse tunnel for <host_id>.<tenant>.teleport.sh found, error dialing peer proxies [<proxy-id>]: error receiving dial response: Teleport proxy failed to connect to "node" agent "@local-node" over reverse tunnel:

  no tunnel connection found: no node reverse tunnel for <host_id>.<tenant>.teleport.sh found

This usually means that the agent is offline or has disconnected. Check the
agent logs and, if the issue persists, try restarting it or re-registering it
with the cluster., error receiving dial response: Teleport proxy failed to connect to "node" agent "@local-node" over reverse tunnel:

  no tunnel connection found: no node reverse tunnel for <host_id>.<tenant>.teleport.sh found

This usually means that the agent is offline or has disconnected. Check the
agent logs and, if the issue persists, try restarting it or re-registering it
with the cluster.

This usually means that the agent is offline or has disconnected. Check the
agent logs and, if the issue persists, try restarting it or re-registering it
with the cluster.
context canceled
```

**Comment by TeleLos** [comment]
Suggested work around would be to add a flag allow you to skip missing servers.

**Comment by nate-thirdwave** [comment]
Any update on this?

**Comment by zmb3** [comment]
This is fixed in Teleport 15 (implemented in #33429)
